### ISSUE TITLE
Pytest fails with `AttributeError` when importing cupy

### ISSUE THREAD

**Issue by shmurai** [chainer/chainer#8138]
We found that `import cupy` in `pytest` can fail with `AttributeError` when the root directory is the Chainer repository:
```
$ cat importcupy.py
import cupy
$ python -m pytest importcupy.py
...
========================================================================== ERRORS ===========================================================================
______________________________________________________________ ERROR collecting importcupy.py _______________________________________________________________
importcupy.py:1: in <module>
    import cupy
../../.pyenv/versions/3.7.3/lib/python3.7/site-packages/cupy/__init__.py:12: in <module>
    from cupy import core  # NOQA
../../.pyenv/versions/3.7.3/lib/python3.7/site-packages/cupy/core/__init__.py:1: in <module>
    from cupy.core import core  # NOQA
cupy/core/carray.pxi:50: in init cupy.core.core
    ???
E   AttributeError: type object 'cupy.core.core.Indexer' has no attribute '__reduce_cython__'
______________________________________________________________ ERROR collecting importcupy.py _______________________________________________________________
importcupy.py:1: in <module>
    import cupy
../../.pyenv/versions/3.7.3/lib/python3.7/site-packages/cupy/__init__.py:12: in <module>
    from cupy import core  # NOQA
../../.pyenv/versions/3.7.3/lib/python3.7/site-packages/cupy/core/__init__.py:1: in <module>
    from cupy.core import core  # NOQA
cupy/core/carray.pxi:50: in init cupy.core.core
    ???
E   AttributeError: type object 'cupy.core.core.Indexer' has no attribute '__reduce_cython__'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
================================================================== 2 error in 5.17 seconds ==================================================================
```

This occurs with master and 406b198, not with a75d9cd.

**Comment by shmurai** [comment]
This is resolved in #8131.

**Comment by niboshi** [comment]
If there's a `conftest.py` with `import cupy`, the error does not occur:

```
$ cat importcupy.py
import cupy
$ cat conftest.py
import cupy
$ python -m pytest importcupy.py
============================= test session starts ==============================
platform linux -- Python 3.6.8, pytest-4.1.0, py-1.7.0, pluggy-0.12.0
rootdir: /home/niboshi/w/repos/chainer, inifile: setup.cfg
plugins: xdist-1.27.0, cov-2.7.1, repeat-0.8.0, forked-1.0.2
collected 0 items

========================= no tests ran in 0.69 seconds =========================
```

**Comment by shmurai** [comment]
How about running actual Chainer tests (e.g. `python -m pytest -s -v -m 'not slow' tests/chainermn_tests/`)? In our environment, this fails with errors related to cupy (this issue is resolved by #8131 as well).

**Comment by niboshi** [comment]
@shmurai Confirmed. But why haven't we noticed that on CIs?

**Comment by niboshi** [comment]
Resolved via #8131
